api/v1beta2/types.go

```diff
@@ -269,7 +269,7 @@ type PortRange struct {
 }
 
 // SecurityGroup defines a VPC Security Group that should exist or be created within the specified VPC, with the specified Security Group Rules.
-// +kubebuilder:validation:XValidation:rule="!has(self.id) && !has(self.name)",message="either an id or name must be specified"
+// +kubebuilder:validation:XValidation:rule="has(self.id) || has(self.name)",message="either an id or name must be specified"
```

Comment:
Could you please help me understand this, How it would affect, User should set either id or name right?

https://playcel.undistro.io/?content=H4sIAAAAAAAAA3VRS0sDMRD%2BK%2BN6aAu1xdeltyJFkapFkSLsZbo73Q0kkyXJti7W%2F%2B5ku6hVSg4Z8mW%2Bx8xHkmPAZJKcwh05AuUhlHJxVQeIkJTwNn2Yg3Vw%2F%2FL0CGvrDIZRyil7ymqnQnPrbF1NUgZQ%2BQTOLy6vrlNOhgm9V468V5ZFYDyGJenMGoJgW5Gb2RwWGptC2jk%2FSVm%2BHL5FO8hiIZDDLKgNCcdq6j2ZlW6gslvxnAPxRjnLhjhEbpHVVrKgEEhNTrXIVoVyr2uNsQyzb3cwRy5qLAj6oj8YRSetmzNYSj6CxtautfaTCEqR7j69emqJ0RGCQLH2Kqc4q1%2BjHB6ZZceyiMzQe66516bYoK7xiDgWqNiHP7vqiGZd%2FtiXWa1JJic9di0MaCpNvjPmK%2BUwYml3SvT9g6WOVD6A3Q7%2BA4yGBsnnFwKmXNFAAgAA

Reply:
Yes, a name or an id must be provided.

Reply:
@Karthik-K-N isn't this appropriate `!has(self.id) && !has(self.name)` ?

Reply:
No, Sometimes CEL expressions are confusing but the main logic is, The message will be displayed when the expression evaluates to false, In the above case if we set `!has(self.id) && !has(self.name)`, it does not allow us to set either one of id or name.
Example when we set id `!true && !false` --> `false && true` --> false = `either an id or name must be specified`

More info: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules